src/portal/src/Portal.js

```diff
@@ -1,38 +1,56 @@
-import { Component } from 'react'
+import { useState, useEffect, memo } from 'react'
 import canUseDom from 'dom-helpers/util/inDOM'
 import ReactDOM from 'react-dom'
 import PropTypes from 'prop-types'
 
 let portalContainer
 
-export default class Portal extends Component {
-  constructor() {
-    super()
+const initializePortal = () => {
+  if (!canUseDom) {
+    return null
+  }
+
+  if (portalContainer) {
+    return portalContainer
+  }
+
+  portalContainer = document.createElement('div')
+  portalContainer.setAttribute('evergreen-portal-container', '')
+  return portalContainer
+}
+
+const initializeEl = () => {
+  if (!canUseDom) {
+    return null
+  }
 
-    // This fixes SSR
-    if (!canUseDom) return
+  return document.createElement('div')
+}
 
-    if (!portalContainer) {
-      portalContainer = document.createElement('div')
-      portalContainer.setAttribute('evergreen-portal-container', '')
+const Portal = memo(({ children }) => {
+  const [portalContainer] = useState(initializePortal())
+  const [el] = useState(initializeEl())
+
+  useEffect(() => {
+    if (portalContainer) {
       document.body.appendChild(portalContainer)
```

Comment:
We only want to add the container to the DOM once. This will add it every time a new Portal is mounted.

Reply:
hmm, what do you think the best way to solve this would be? Simple DOM check to see if it exists before appending?

Reply:
Move this to the `initializeContainer` function only when `portalContainer` isn't already defined.